provide.tests/VaultTest.cs

```diff
@@ -103,5 +106,48 @@ public async void TestCreateSecret()
             );
             Assert.NotNull(secret.Id);
         }
+
+        [Fact]
+        public async void TestSignUsingHdWallet()
+        {
+            var token = await TestUtil.CreateIdentForTestUser();
+            var ident = Ident.InitIdent(token);
+            var org = await ident.CreateOrganization(new Model.Ident.Organization
+            {
+                Name = "test org"
+            });
+            var orgToken = await ident.CreateToken(new Model.Ident.JWTToken { OrganizationId = org.Id });
+
+            var vlt = Vault.InitVault(orgToken.Token);
+            var vault = await vlt.CreateVault(new provide.Model.Vault.Vault
+            {
+                Name = "test vault"
+            });
+
+            var hdWallet = await CreateVaultKey(vlt, vault.Id.ToString(), "BIP39", "hdwallet", "EthHdWallet");
+
+            var payload = new byte[3] { 1, 2, 3 };
+            var message = Encoding.UTF8.GetString(payload);
+            var signedMessage = await vlt.SignMessage(vault.Id.ToString(), hdWallet.Id.ToString(), message);
+
+            Assert.NotNull(signedMessage.Signature);
+            Assert.NotEmpty(signedMessage.Signature);
+
+            var verifiedMessage = await vlt.VerifySignature(vault.Id.ToString(), hdWallet.Id.ToString(), message, signedMessage.Signature);
+            // this assert fails

```

Comment:
See https://docs.provide.services/vault/api-reference/keys/signing-and-verification#verification-options-bip-39